src/Download.php

```diff
@@ -0,0 +1,169 @@
+<?php
+
+namespace Storj\Uplink;
+
+use DateTimeImmutable;
+use Exception;
+use FFI;
+use FFI\CData;
+use Generator;
+use IteratorAggregate;
+use Psr\Http\Message\StreamInterface;
+use Storj\Uplink\Exception\IOException;
+use Storj\Uplink\Exception\UplinkException;
+use Storj\Uplink\Internal\Scope;
+use Storj\Uplink\Internal\Util;
+use Storj\Uplink\Psr\ReadStream;
+
+/**
+ * Handle to a remote object
+ *
+ * TODO: can the download be read only once? Should we protect against reading it twice?
+ */
+class Download
+{
+    /**
+     * We seem to receive 7408 bytes per read
```

Comment:
Does this mean you never read more than 7408 bytes on a single read regardless of the chunk size?

Reply:
Yes that is what I observe

Reply:
If this is intentional we may just leave out the chunksize from the interface.

Reply:
This sounds like something we need to investigate in the Golang or C library.